kde-apps/kdepim-addons/kdepim-addons-9999.ebuild

```
@@ -74,11 +89,39 @@ src_configure() {
 		-DOPTION_USE_PLASMA_ACTIVITIES=$(usex activities)
 		$(cmake_use_find_package importwizard KPim6ImportWizard)
 		$(cmake_use_find_package markdown Discount)
+		-DKDEPIM_RUN_AKONADI_TEST=OFF # tests need database software and networking
 	)
 
 	ecm_src_configure
 }
 
+src_test() {
+	local dbus_params=(
+		$(dbus-daemon --session --print-address --fork --print-pid)
+	)
+	local -x DBUS_SESSION_BUS_ADDRESS=${dbus_params[0]}
+
+	local CMAKE_SKIP_TESTS=(
+		# Locale differences in date display.
+		"fancyheaderstyleplugintest"
+		"grantleeheaderstyleplugintest"
+		# Comparison files outdated, also affected by changes in other packages.
+		"messageviewerplugins-rendertest"
+		# Tests pass but segfault when they exit.
+		"kdepim-addons-eventedittest"
+		"messageviewer-dkimauthenticationverifiedserverdialogtest"
+		# Test pass but get stuck indefinetly afterwards.
+		"kdepim-addons-todoedittest"
+	)
+
+	local -x QT_QPA_PLATFORM=offscreen
+
+	# tests can get stuck with spawned processes, 4 minutes is a reasonable timeout
+	cmake_src_test --timeout $(( 60 * 4 )) # seconds
```

```suggestion
	# tests can get stuck with spawned processes, 4 minutes is a reasonable timeout
	ecm_src_test --timeout $(( 60 * 4 )) # seconds
```

ecm.eclass exports src_test, but also gives you `QT_QPA_PLATFORM=offscreen` for free